test/e2e/node_test.go

```diff
@@ -233,17 +233,20 @@ func (k *NodeTests) SameBuildingBlock(t *testing.T) {
 
 	var s []struct {
 		BuildingBlock string `json:"OS-EXT-SRV-ATTR:host"`
+		ID            string `json:"id"`
 	}
 	err = servers.ExtractServersInto(allPages, &s)
 	require.NoError(t, err, "There should be no error extracting server info")
 
+	require.Len(t, s, k.ExpectedNodeCount, "Expeted to find %d building blocks", k.ExpectedNodeCount)
+
 	bb := ""
-	for _, bbs := range s {
+	for i, bbs := range s {
 		require.NotEmpty(t, bbs.BuildingBlock, "Node building block should not be empty")
 		if bb == "" {
-			bb = string(bbs.BuildingBlock)
+			bb = bbs.BuildingBlock
 		} else {
-			require.Equal(t, bb, bbs.BuildingBlock, "Nodes should be on the same building block")
+			require.Equal(t, bb, bbs.BuildingBlock, "Node %s is on building block %s which is different from node %s which is running on %s", bbs.ID, bbs.BuildingBlock, s[i-1].ID, s[i-1].BuildingBlock)
```

Comment:
I we have more than two nodes it is not necessarily the previous one, isn't it? I think you should compare to the first one.

Reply:
I could do that, sure. But the way the loop/check is designed it will break/fail as soon as the first different building block is found, so previous one will always be different. But your suggestion is a little bit simpler